blaze/expr/strings.py

```diff
@@ -165,13 +296,74 @@ def isstring(ds):
 _mul, _rmul = _mkbin('mul', Repeat)
 
 
+class str_ns(object):
+
+    def __init__(self, field):
+        self.field = field
+
+    def upper(self):
+        return str_upper(self.field)
+
+    def lower(self):
+        return str_lower(self.field)
+
+    def len(self):
+        return str_len(self.field)
+
+    def like(self, pattern):
+        return like(self.field, pattern)
+
+    def cat(self, other, sep=None):
+        return str_cat(self.field, other, sep=sep)
+
+    def find(self, sub):
+        return str_find(self.field, sub)
+
+    def isalnum(self): return str_isalnum(self.field)
+    def isalpha(self): return str_isalpha(self.field)
+    def isdecimal(self): return str_isdecimal(self.field)
+    def isdigit(self): return str_isdigit(self.field)
+    def islower(self): return str_islower(self.field)
+    def isnumeric(self): return str_isnumeric(self.field)
+    def isspace(self): return str_isspace(self.field)
+    def istitle(self): return str_istitle(self.field)
+    def isupper(self): return str_isupper(self.field)
+
+    def replace(self, old, new, max=None):
+        return str_replace(self.field, old, new, max)
+
+    def capitalize(self):
+        return str_capitalize(self.field)
+
+    def pad(self, width, side=None, fillchar=None):
+        return str_pad(self.field, width, side, fillchar)
+
+    def strip(self): return str_strip(self.field)
+    def lstrip(self): return str_lstrip(self.field)
+    def rstrip(self): return str_rstrip(self.field)
+
+    def __getitem__(self, idx):
+        return str_slice(self.field, idx)
+
+    def slice_replace(self, start=None, stop=None, repl=None):
+        return str_slice_replace(self.field, start, stop, repl)
+
+class str(object):
+
+    __name__ = 'str'
+
+    def __get__(self, obj, type=None):
```

Comment:
> if obj is none we should return self

+1

>  type doesn't need a default argument.

I thought that was part of the [descriptor protocol](https://docs.python.org/3.5/howto/descriptor.html#descriptor-protocol).